packages/tools/src/user-interaction.ts

```diff
@@ -0,0 +1,86 @@
+import { confirm, select, text } from "@clack/prompts";
+import { tool } from "ai"
+import { z } from "zod"
+
+type Spinner = {
+    start: (msg?: string) => void;
+    stop: (msg?: string, code?: number) => void;
+    message: (msg?: string) => void;
+}
+
+export const makeInteractiveToolset = (s: Spinner) => {
+
+    return {
+        askQuestion: askQuestion(s),
+        getConfirmation: getConfirmation(s),
+        presentOptions: presentOptions(s),
+        concludeTask
+    }
+}
+
+export const askQuestion = (s: Spinner) =>
+    tool({
+        description: 'Interact with user to as him a question',
+        parameters: z.object({
+            question: z.string().describe("What do you want to ask"),
+        }),
+        execute: async ({ question }) => {
+            s.stop("A question:");
```

Comment:
Spinner is not necessary here. We should get back to using `onStepStart` for noisy commands (the ones that produce some sort of of output) and hopefully, it gets merged soon, or we just figure a better way for all commands. This tool ideally should just ask via the terminal.

Reply:
There is no "onStepStart". Guess they removed it. (that's why i changed the implementation)

Reply:
I am thinking about making a "interactiveToolMaker" that adds automatically spinner starts and stops. Or something like that.

Reply:
There was never `onStepStart`, but it's here via patch-package. I suppose you're not using `bun`, which did not install it for you! Nevertheless, I'll play around with it. If that's the only request, I will send merge the PR with updates accordingly.